src/AMDGPU.jl

```diff
@@ -123,7 +123,10 @@ function __init__()
                 librocfft !== nothing      && include(joinpath(@__DIR__, "fft", "rocFFT.jl"))
                 #librocsparse !== nothing  && include("sparse/rocSPARSE.jl")
                 #librocalution !== nothing && include("solver/rocALUTION.jl")
-                #librocrand !== nothing    && include("rand/rocRAND.jl")
+                if librocrand !== nothing
```

Comment:
Nitpick: If these could be merged into a single `include`, that would be a bit more consistent.

Reply:
I was thinking more that the file can be included in `src/rand/rocRAND.jl` since it depends on rocRAND. Not a big deal though, I might just change it later :slightly_smiling_face:

Reply:
Ah, ok. But then the function defined there (e.g. `seed!`) would not be available as `AMDGPU.seed!`, or could this still be done?

Reply:
If that's the case, then you could split out the code that doesn't use rocRAND directly to `src/random.jl`.

Reply:
Ok, I will think about a good way to do this.
However, we then also need to tests which functions work with the GPUArrays random number generator.

How should we handle e.g. `AMDGPU.seed!`, since it should seed both rngs, if rocrand is available, and only the GPUArrays one otherwise?
Should I just define `rocrand_rng` in interface.jl to be the same as `gpuarrays_rng` if rocrand is not available?

Reply:
> Should I just define `rocrand_rng` in interface.jl to be the same as `gpuarrays_rng` if rocrand is not available?

It depends on the difference in supported functionality between rocrand and gpuarrays RNG; if GPUArrays supports about the same amount of calls as rocRAND, then yes, just default to GPUArrays.